--- xf86-video-sis-0.10.7/src/sis_driver.c.orig	2013-04-07 21:43:38.806080625 +0200
+++ xf86-video-sis-0.10.7/src/sis_driver.c	2013-04-07 21:45:02.936957482 +0200
@@ -426,17 +426,16 @@
 SISErrorLog(ScrnInfoPtr pScrn, const char *format, ...)
 {
     va_list ap;
-    static const char *str = "**************************************************\n";
 
     va_start(ap, format);
-    xf86DrvMsg(pScrn->scrnIndex, X_ERROR, str);
+    xf86DrvMsg(pScrn->scrnIndex, X_ERROR, "**************************************************\n");
     xf86DrvMsg(pScrn->scrnIndex, X_ERROR,
 	"                      ERROR:\n");
     xf86VDrvMsgVerb(pScrn->scrnIndex, X_ERROR, 1, format, ap);
     va_end(ap);
     xf86DrvMsg(pScrn->scrnIndex, X_ERROR,
 	"                  END OF MESSAGE\n");
-    xf86DrvMsg(pScrn->scrnIndex, X_ERROR, str);
+    xf86DrvMsg(pScrn->scrnIndex, X_ERROR, "**************************************************\n");
 }
 
 static void
@@ -1785,7 +1784,6 @@
     DisplayModePtr currentMode, firstMode;
     Bool infochanged = FALSE;
     Bool usenonrect = pSiS->NonRect;
-    const char *rectxine = "\t... setting up rectangular Xinerama layout\n";
 
     pSiS->MBXNR1XMAX = pSiS->MBXNR1YMAX = pSiS->MBXNR2XMAX = pSiS->MBXNR2YMAX = 65536;
     pSiS->HaveNonRect = pSiS->HaveOffsRegions = FALSE;
@@ -1914,14 +1912,14 @@
        if(infochanged && !usenonrect) {
 	  xf86DrvMsg(pScrn1->scrnIndex, X_INFO,
 			"Virtual screen size does not match maximum display modes...\n");
-	  xf86DrvMsg(pScrn1->scrnIndex, X_INFO, rectxine);
+	  xf86DrvMsg(pScrn1->scrnIndex, X_INFO, "\t... setting up rectangular Xinerama layout\n");
 
        }
     } else if(infochanged && usenonrect) {
        usenonrect = FALSE;
        xf86DrvMsg(pScrn1->scrnIndex, X_INFO,
 		"Only clone modes available for this virtual screen size...\n");
-       xf86DrvMsg(pScrn1->scrnIndex, X_INFO, rectxine);
+       xf86DrvMsg(pScrn1->scrnIndex, X_INFO, "\t... setting up rectangular Xinerama layout\n");
     }
 
     if(pSiS->maxCRT1_X1) {		/* Means we have at least one non-clone mode */
