--- xf86-video-sis-0.10.9/src/sis_driver.c~	2017-01-18 07:45:12.000000000 +0900
+++ xf86-video-sis-0.10.9/src/sis_driver.c	2018-05-23 22:17:33.239096854 +0900
@@ -2738,7 +2738,7 @@
     float hsync, refresh = 0.0;
     char *desc, *desc2, *prefix, *uprefix, *output;
 
-    xf86DrvMsg(pScrn->scrnIndex, pScrn->virtualFrom, "Virtual size is %dx%d "
+    xf86DrvMsg(pScrn->scrnIndex, X_CONFIG, "Virtual size is %dx%d "
 	       "(pitch %d)\n", pScrn->virtualX, pScrn->virtualY,
 	       pScrn->displayWidth);
 
